### ISSUE TITLE
Bug: [no-invalid-void-type] should report `accessor` properties with an invalid `void` type

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10843]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEAqCmEAuBGaBvAUNaB6X0ATvAA4D2hS8AJtMBccEiAJ7bQmFkkByYAtvABc0AG5kAltQDcmAL6ZMoSDATIATBnb5oEABZkAriFoAjeEVIUqZw0mjUyiAHYByJOzDBgiCBQ5cvALCYpIy8phAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDcBDRgBNaPDpWFFS6KImjQO0SODABfECqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Test1 {
  // reported correctly
  propName: void;
}

class Test2 {
  // should be reported but doesn't
  accessor propName: void;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-invalid-void-type": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected the rule to report on the `accessor` property, similar to how it reports the standard property.

### Actual Result

The `accessor` property didn't get reported.

### Additional Info

Related: https://github.com/typescript-eslint/typescript-eslint/issues/10763#issuecomment-2634900340, one more `accessor` issue 😅

**Comment by JoshuaKGoldberg** [comment]
🫠 so many accessor bugs...
